### DISCUSSION TITLE
How to  add Label to the Triangle lines?

### DISCUSSION THREAD

**Discussion by blackjyn** [typst/typst#3494] [Q&A] [answered]
So I drew a simple triangle using `polygon` fx

```typst
#polygon(
  (50%, 0cm),
  (50%, 4cm),
  (0%, 4cm),  
)
```
it renders like this:
![ScreenShot Tool -20240226090130](https://github.com/typst/typst/assets/6046191/4b977109-63e1-4079-a865-f4a3213a9745)

now how do I add label to the lines so it rendered like this:

![ScreenShot Tool -20240226090130 (1)](https://github.com/typst/typst/assets/6046191/e4bb5ef7-5fd7-4a6e-b8e5-8b537b59bc9f)

thank you!

**Comment by supcik** [comment, marked as answer]
I would do it with a `box` and some `place` :

```
#box(
polygon(
  (50%, 0cm),
  (50%, 4cm),
  (0%, 4cm),  
)
+ place(dx: 35mm, dy: -30mm, text(size: 24pt, fill: orange, [a]))
+ place(dx: 45mm, dy: 4mm, text(size: 24pt, fill: orange, [b]))
+ place(dx: 84mm, dy: -24mm, text(size: 24pt, fill: orange, [c]))
)
```

**Comment by supcik** [comment]
![image](https://github.com/typst/typst/assets/30344/90864273-6787-433e-94f2-100a9a05a65b)

**Reply by supcik** [reply]
You can change the `dy` of "b" if you want to have it inside the triangle

**Comment by blackjyn** [comment]
Thank you so much @supcik !

**Reply by supcik** [reply]
You are welcome.
